component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordChangeEnforcerOnExpiration.java

```diff
@@ -233,7 +234,12 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                     log.debug("Updated user credentials of " + tenantAwareUsername);
                 }
             } catch (org.wso2.carbon.user.core.UserStoreException e) {
-                throw new AuthenticationFailedException("Incorrect current password", e);
+                if(e.getCause() instanceof PrivilegedActionException) {
```

Comment:
Better to use 
catch(PrivilegedActionException .) {
..
} catch (UserStoreException .) { }

Reply:
The updateCredentials method is throwing only the UserStoreException. If I use catch blocks as you said I'm getting error "Exception PrivilegedActionException is never thrown in the corresponding try block".